### ISSUE TITLE
Lack of typing

### ISSUE THREAD

**Issue by antoine-pous** [HGData/mk-node-hubspot#181]
Actually this lib do a great job but the lack of typing for most of `opts` parameters is really painful.

We have to check regularly HubSpot APIs doc to know which fields/values are accepted.

I would like help to improve this part but i don't really know which way will be better, so i would like having some advices:

1. Full refactor to move the whole codebase to TypeScript, following the latests standards
2. Improve the current types file which can become really painful instead of maintaining the TypeScript codebase

I think the one is the best thing to do before this module become too big, actually moving to TypeScript shouldn't be so long and can ensure the code quality.

**Comment by pcothenet** [comment]
@antoine-pous I agree, 1 would be the best approach. Keeping separate types up-to-date is a pain.

**Comment by pcothenet** [comment]
@antoine-pous did you ever get to try to migrate?

**Comment by pcothenet** [comment]
@vkhomiv-hubspot what's your take on this? We've had half-a-dozen issues related to typing. They're always annoying to maintain when generated on the side. My inclination (since I do mostly typescript these days) would be to migrate the full library to typescript, but wondering what you think?

**Comment by antoine-pous** [comment]
@pcothenet i've created a typescript branch, actually i don't really made changes, just migrate the codebase to TS step by step

**Comment by vkhomiv-hubspot** [comment]
I agree it's a big deal.

**Comment by phips28** [comment]
@antoine-pous any update on this?
